tsl/src/data_node.c

```diff
@@ -552,16 +552,26 @@ connect_for_bootstrapping(const char *node_name, const char *const host, int32 p
 						  const char *username, const char *password)
 {
 	TSConnection *conn = NULL;
+	char *err = NULL;
 	int i;
+
 	for (i = 0; i < lengthof(bootstrap_databases); i++)
 	{
 		List *node_options =
 			create_data_node_options(host, port, bootstrap_databases[i], username, password);
-		conn = remote_connection_open_with_options_nothrow(node_name, node_options);
+		conn = remote_connection_open_with_options_nothrow(node_name, node_options, &err);
+
 		if (conn)
 			return conn;
 	}
-	return conn;
+
+	if (NULL == conn)
+		ereport(ERROR,
+				(errcode(ERRCODE_SQLCLIENT_UNABLE_TO_ESTABLISH_SQLCONNECTION),
+				 errmsg("could not connect to \"%s\"", node_name),
+				 err == NULL ? 0 : errdetail("%s", err)));
```

Comment:
Copy-pasted code with `connection.c`. I don't think it is important to improve now and I don't see an easy and clear pattern to fix.